mllib/src/main/scala/org/apache/spark/ml/source/image/ImageFileFormat.scala

```diff
@@ -49,6 +49,10 @@ private[image] class ImageFileFormat extends FileFormat with DataSourceRegister
 
   override def shortName(): String = "image"
 
+  override def hashCode(): Int = getClass.getCanonicalName.hashCode()
+
+  override def equals(other: Any): Boolean = other.isInstanceOf[ImageFileFormat]
```

Comment:
Hm, the options can be different. For example, `dropInvalid` can be enabled/disabled, which results in different outputs. Is this safe to assume that they are same?

Reply:
The options is not in the instance of `ImageFileFormat`

Reply:
Ah okie dokie